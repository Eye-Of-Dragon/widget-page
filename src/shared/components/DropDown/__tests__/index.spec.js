import React from 'react';
import Dropdown from '../index';
import { render } from '@testing-library/react';

jest.mock(
  "../../../../../node_modules/antd/lib/select",
  (): any => {

    const Select = ({ children, onChange }) => {
      return <select onChange={e => onChange(e.target.value)}>{children}</select>;
    };

    Select.Option = ({ children, ...otherProps }) => {
      return <option {...otherProps}>{children}</option>;
    };

    return Select;
  },
);

describe('Dropdown', () => {

  it('snap-shoot for DropDown', () => {
    const options = {
      male: 'Male',
      female: 'Female',
    };
    const { container } = render(<Dropdown
        lebel={'Sexe'}
        name={'sexe'}
        options={options}
      />);
    expect(container).toMatchSnapshot();
  });

  it('Drop Down get size of options', () => {
    const options = {
      male: 'Male',
      female: 'Female',
    };
    const { container } = render(<Dropdown
      lebel={'Sexe'}
      name={'sexe'}
      options={options}
    />);
    const optionElements = container.getElementsByTagName('option');
    expect(optionElements.length).toBe(2)
  });
});
