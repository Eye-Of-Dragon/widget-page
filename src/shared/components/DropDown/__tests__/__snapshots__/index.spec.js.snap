// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown snap-shoot for DropDown 1`] = `
<div>
  <select>
    <option
      value="male"
    >
      Male
    </option>
    <option
      value="female"
    >
      Female
    </option>
  </select>
</div>
`;

exports[`Dropdown snapShoot for DropDown 1`] = `
<div>
  <select>
    <option
      value="male"
    >
      Male
    </option>
    <option
      value="female"
    >
      Female
    </option>
  </select>
</div>
`;
